Source/control.cpp

```diff
@@ -2078,7 +2078,7 @@ char *control_print_talk_msg(char *msg, int *x, int y, int color)
 
 		c = fontframe[gbFontTransTbl[(BYTE)*msg]];
 		width += fontkern[c] + 1;
-		if (width > 514 + PANEL_LEFT)
+		if (width > (450 + SCREEN_X) + PANEL_LEFT)

```

Comment:
we have `PANEL_X` for where the panel position needs to be relative to the buffer and not the screen.